tests/components/oculus-touch-controls.test.js

```diff
@@ -10,38 +10,51 @@ suite('oculus-touch-controls', function () {
     el.setAttribute('oculus-touch-controls', '');
     el.addEventListener('loaded', function () {
       component = el.components['oculus-touch-controls'];
-      component.controllersWhenPresent = [
-        {id: 'Oculus Touch', index: 0, hand: 'left', pose: {}}
-      ];
+      // Initially no controllers are present
+      component.controllers = [];
+      // Our Mock data for enabling the controllers.
+      component.controllersWhenPresent = [{
+        id: 'Oculus Touch',
+        index: 0,
+        hand: 'left',
+        pose: {}
+      }];
       done();
     });
   });
 
   suite('checkIfControllerPresent', function () {
-    test('removes event listeners if controllers not present', function () {
-      var addEventListenersSpy = sinon.spy(component, 'addEventListeners');
-      var injectTrackedControlsSpy = sinon.spy(component, 'injectTrackedControls');
-      var removeEventListenersSpy = sinon.spy(component, 'removeEventListeners');
-
-      // Mock has not been checked previously.
-      delete component.controllerPresent;
-
-      component.checkIfControllerPresent();
-
-      sinon.assert.notCalled(injectTrackedControlsSpy);
-      sinon.assert.notCalled(addEventListenersSpy);
-      sinon.assert.calledOnce(removeEventListenersSpy);
-      assert.strictEqual(component.controllerPresent, false);
+    var component;
+    var controllerSystem;
+    var addEventListenersSpy;
+    var injectTrackedControlsSpy;
+    var removeEventListenersSpy;
+
+    setup(function (done) {
+      component = this.el.components['oculus-touch-controls'];
+      controllerSystem = this.el.sceneEl.systems['tracked-controls-webvr'];
+      addEventListenersSpy = sinon.spy(component, 'addEventListeners');
+      injectTrackedControlsSpy = sinon.spy(component, 'injectTrackedControls');
+      removeEventListenersSpy = sinon.spy(component, 'removeEventListeners');
+      done();
     });
 
-    test('does not call removeEventListeners multiple times', function () {
```

Comment:
Same for this. Is this test intentionally removed?

Reply:
This is redundant with the base state that removeEventListeners is not called if the controller is not present. This is covered by test 'returns not present if no controllers on first call'. This is also the same set of tests for oculus-go-controls. This is meant to standardize down to the correct set of compliant tests for these components so they can be updated more easily, specifically, these improvements are meant to make adding Quest controller support easier (I'm going to upgrade the oculus-touch-controls for this since Quest controllers are really just generation 2 of Oculus Touch and share nearly all of the same behavior except for the models).

Reply:
I like the idea of consolidating Quest / Rift to `oculus-controls` but two reasons against:

- Buttons animation: There's code in the component that is model dependent and would have to be duplicated making the component more complicated. Debugging, bug diagnosis, maintenance will be harder.
- Tool tips and instructions: Separate components make easier and consistent with other controllers for an application to discriminate and display instructions and tooltips. 

At the moment, I favor keeping separate `oculus-touch-controls` and `oculus-quest-controls` even though there's some not ideal code and test duplication.